### PR TITLE
fix(cdk): `TuiDialogHostComponent` opening many different types dialogs together

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [x] Bugfix
-   [ ] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/cdk/components/dialog-host/dialog-host.component.ts [hunk 1]
```diff
@@ -13,10 +14,12 @@ import {merge, Observable} from 'rxjs';
     animations: [TUI_PARENT_ANIMATION],
 })
 export class TuiDialogHostComponent<T extends TuiAriaDialogContext> {
-    readonly dialogs$ = merge(...this.dialogs);
+    readonly dialogs$ = combineLatest(this.allTypesDialogs).pipe(
+        map(allTypesDialogs => new Array<T>().concat(...allTypesDialogs)),
```
Comment (nsbarsukov) [inline projects/cdk/components/dialog-host/dialog-host.component.ts:18]: i've understood right now that it is bad solution: it doesn't take order of dialog opening into consideration =/

I move PR to draft and think more

Comment (nsbarsukov) [reply projects/cdk/components/dialog-host/dialog-host.component.ts]: fixed

File: projects/cdk/abstract/dialog.service.ts [hunk 2]
```diff
@@ -37,6 +37,7 @@ export abstract class AbstractTuiDialogService<T extends {}> extends Observable<
                 completeWith,
                 $implicit: observer,
                 component: this.component,
+                createdAt: Date.now(),
```
Comment (waterplea) [inline projects/cdk/abstract/dialog.service.ts:40]: Maybe we can just use id below? It's autogenerated based on ever increasing increment and `Date.now()`

File: projects/cdk/components/dialog-host/dialog-host.component.ts [hunk 3]
```diff
@@ -13,10 +14,16 @@ import {merge, Observable} from 'rxjs';
...
+            new Array<T>()
+                .concat(...allTypesDialogs)
+                .sort((a, b) => a.createdAt - b.createdAt),
+        ),
+    );
 
     constructor(
         @Inject(TUI_DIALOGS)
-        private readonly dialogs: ReadonlyArray<Observable<ReadonlyArray<T>>>,
+        private readonly allTypesDialogs: Observable<readonly T[]>[],
```
Comment (waterplea) [inline projects/cdk/components/dialog-host/dialog-host.component.ts:27]: Let's rename it to `dialogsByType` if you want to emphasize these are arrays of different type dialogs.

File: projects/cdk/abstract/dialog.service.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/cdk/abstract/dialog.service.ts:40]: i'm not sure that it is good idea.

If I use id I will need to remove `tui_` prefix from id (before converting to the number).

Moreover, does it violate the principles of Dependency Injection if component knows how  TuiIdService generate id?
TuiIdService can be changed in future that will break logic inside dialogs.

File: projects/cdk/components/dialog-host/dialog-host.component.ts (hunk 3 above)
Comment (nsbarsukov) [reply projects/cdk/components/dialog-host/dialog-host.component.ts]: done

**Review by waterplea — APPROVED** [review]

**Pull request merged**